From bea1f53ae2ba7608503051b874db9aecb97d4f00 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 18 Jun 2020 09:39:23 +0800
Subject: [PATCH] policy/modules/system/systemd: systemd-user-sessions: make
 systemd_sessions_t MLS trusted for reading/writing from files at all levels

Fixes:
avc:  denied  { search } for  pid=229 comm="systemd-user-se"
name="journal" dev="tmpfs" ino=10956
scontext=system_u:system_r:systemd_sessions_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0
avc:  denied  { write } for  pid=229 comm="systemd-user-se" name="kmsg"
dev="devtmpfs" ino=10032
scontext=system_u:system_r:systemd_sessions_t:s0-s15:c0.c1023
tcontext=system_u:object_r:kmsg_device_t:s15:c0.c1023 tclass=chr_file
permissive=0

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 1449d2808..6b0f52d15 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -1125,6 +1125,8 @@ seutil_read_file_contexts(systemd_sessions_t)
 
 systemd_log_parse_environment(systemd_sessions_t)
 
+mls_file_read_to_clearance(systemd_sessions_t)
+mls_file_write_all_levels(systemd_sessions_t)
 
 #########################################
 #
-- 
2.17.1

